apiserver/pinger_test.go

```diff
@@ -149,18 +148,11 @@ func waitForClock(c *gc.C, clock *testing.Clock) {
 }
 
 func checkConnectionDies(c *gc.C, conn api.Connection) {
-	attempt := utils.AttemptStrategy{
-		Total: coretesting.LongWait,
-		Delay: coretesting.ShortWait,
-	}
-	for a := attempt.Start(); a.Next(); {
-		err := pingConn(conn)
-		if err != nil {
-			c.Assert(err, gc.ErrorMatches, "connection is shut down", gc.Commentf("details: %s", errors.Details(err)))
-			return
-		}
+	select {
+	case <-conn.Broken():
```

Comment:
I'll add a test to the api package, where it belongs. The Ping method is documented as being deprecated, and I'm not sure what holds us back from just removing it; but I do think we should have a test until it is removed.